src/extlibs/hrvo/hrvo_test.cpp

```diff
@@ -173,86 +211,192 @@ class HRVOTest : public ::testing::Test
 
             auto finish_tick_time = std::chrono::high_resolution_clock::now();
             computation_time += finish_tick_time - start_tick_time;
-        } while (!simulator.haveReachedGoals() && prev_frame_time < 15.f);
+        } while (!simulator.haveReachedGoals() && prev_frame_time < simulation_timeout);
 
         auto finish_time = std::chrono::high_resolution_clock::now();
         std::chrono::duration<double> total_time = finish_time - start_time;
         std::cout << "Total run time = " << total_time.count() << std::endl;
         std::cout << "Total simulation time = " << prev_frame_time << std::endl;
-        std::cout << "Average time per tick = " << total_time.count() / frame
-                  << std::endl;
+        std::cout << "Avg time per tick = " << total_time.count() / frame << std::endl;
 
         output_file.close();
         std::cout << "Test information outputted to " << output_file_loc << std::endl;
     }
 };
 
-TEST_F(HRVOTest, div_b_edge_test)
+TEST_F(HRVOTest, stationary_friendly_robot_dodging_moving_friendly_robot)
 {
-    const Vector2 goal_offset  = Vector2(8.f, 0);
-    const Vector2 robot_offset = Vector2(0.f, -ROBOT_RADIUS * 2.5f);
-    for (std::size_t i = 0; i < 1; ++i)
-    {
-        const Vector2 position = -1 * (goal_offset / 2) + Vector2(0.f, 2.5f) +
-                                 (static_cast<float>(i) * robot_offset);
-        simulator.addAgent(position, simulator.addGoal(position + goal_offset));
-    }
-    add_static_obstacle(Vector2(0, 2.f), 0.75f);
-    create_div_b_field();
+    std::vector<std::pair<Point, Point>> friendly_start_dest_points = {
+        std::pair(Point(0.0, 0.0), Point(0.0, 0.0)),
+        std::pair(Point(-2.0, 0.0), Point(4.0, 0.0))};
+    instantiate_robots_in_world(friendly_start_dest_points, {});
+}
+
+TEST_F(HRVOTest, stationary_friendly_robot_dodging_moving_enemy_robot)
```

Comment:
It looks like the robots collide, when will there be validation for these tests?

Reply:
These tests serve as an easy way to visualize various scenarios, such as edge cases, so I did not add any validation. It would probably be better to have actual validation once HRVO is running on the ERForce Sim with the correct field and robot configuration, rather than the simplified HRVO simulator. That way we can also tune any values/formulas with a more realistic simulator.
About the robots colliding, I think the expected behavior would be that friendly robots do not dodge enemy robots once they have reached their destination. Though I haven't thought about it too much, so it's open for debate.